cmd/blocking-issue-creator/main.go

```diff
@@ -87,187 +104,77 @@ func main() {
-				logger.WithError(err).Error("Failed to update issue.")
-				failed = true
-				return nil
-			}
+	for _, branch := range branches.List() {
+		body += fmt.Sprintf(" - `%s`\n", branch)
+		branchTokens = append(branchTokens, fmt.Sprintf("branch:%s", branch))
+	}
+	body += "\nContact the [Test Platform](https://coreos.slack.com/messages/CBN38N3MW) or [Automated Release](https://coreos.slack.com/messages/CB95J6R4N) teams for more information."
+	title := fmt.Sprintf("Future Release Branches Frozen For Merging | %s", strings.Join(branchTokens, " "))
 
-			logger.Infof("Updated issue %d", updateIssue.UpdateIssue.Issue.Number)
-		} else {
-			// we need to create a new issue
-
-			// what is the ID of the blocker label?
-			var labelQuery struct {
-				Repository struct {
-					ID    githubql.ID
-					Label struct {
-						ID githubql.ID
-					} `graphql:"label(name:\"tide/merge-blocker\")"`
-				} `graphql:"repository(owner: $owner, name: $name)"`
-			}
-			vars = map[string]interface{}{
-				"owner": githubql.String(repoInfo.Org),
-				"name":  githubql.String(repoInfo.Repo),
-			}
-			if err := client.Query(context.Background(), &labelQuery, vars); err != nil {
-				logger.WithError(err).Error("Failed to search for merge blocker labels.")
-				failed = true
-				return nil
-			}
-			if labelQuery.Repository.Label.ID == nil {
-				logger.Error("Failed to find a merge blocker label.")
-				failed = true
-				return nil
-			}
+	query := fmt.Sprintf("is:issue state:open label:\"tide/merge-blocker\" repo:%s/%s author:%s", repoInfo.Org, repoInfo.Repo, githubLogin)
+	sort := "updated"
+	// We will make sure that the first issue in the list will be with the most recent update.
+	ascending := false
 
-			var createIssue struct {
-				CreateIssue struct {
-					Issue struct {
-						Number githubql.Int
-					}
-				} `graphql:"createIssue(input: $input)"`
-			}
-			// this needs to be a named type for the library
-			type CreateIssueInput struct {
-				RepositoryID githubql.ID     `json:"repositoryId"`
-				Title        githubql.String `json:"title"`
-				Body         githubql.String `json:"body"`
-				LabelIDs     []githubql.ID   `json:"labelIds"`
-			}
-			input := CreateIssueInput{
-				RepositoryID: labelQuery.Repository.ID,
-				Title:        githubql.String(title),
-				Body:         githubql.String(body),
-				LabelIDs:     []githubql.ID{labelQuery.Repository.Label.ID},
-			}
+	issues, err := client.FindIssues(query, sort, ascending)
+	if err != nil {
+		logger.WithError(err).Error("Failed to search for open issues.")
+		return err
+	}
 
-			if !o.Confirm {
-				logger.Info("Would create issue.")
-				return nil
+	if len(issues) > 1 {
+		logger.Warnf("Found more than one merge blocking issue by the bot: %v", len(issues))
+		for _, issue := range issues[1:] {
+			if err := client.CloseIssue(repoInfo.Org, repoInfo.Repo, issue.Number); err != nil {
+				logger.WithError(err).Error("Failed to close issue.")
+				return err
```

Comment:
For instance, your refactor makes this flow change. Now, we fail and exit early if we can't close an issue. That is not intended.